doc/Language/syntax.pod6

```diff
@@ -223,109 +223,193 @@ say 'code again';
+
+=head3 Ordinary identifiers
+
+An ordinary identifier is composed of a leading alphabetic character
+that may be followed by one or more alphanumeric characters. It may also
+contain isolated, embedded apostrophes C<'> and/or hyphens C<->, provided
+that the next character is each time alphabetic.
+
+The definitions of "alphabetic" and "alphanumeric" include appropriate Unicode
+characters. Which characters are "appropriate" depends on the implementation.
+In the Rakudo/MoarVM Perl 6 implementation alphabetic characters include
+Unicode characters in the character category I<Letter>, and the underscore
+C<_>. Alphanumeric characters additionally include Unicode characters in the
+general category I<Number, Decimal Digit> (Nd).
 
 =begin code :skip-test
-# valid identifiers:
+# valid ordinary identifiers:
 x
 something-longer
 with-numbers1234
-don't
-fish-food
-π
+don't-do-that
+piece_of_π
+駱駝道                 # "Rakuda-dō", Japanese for "Way of the camel"
 =end code
 
 =begin code :skip-test
-# not valid identifiers:
-with-numbers1234-5
-42
-is-prime?
-fish--food
+# invalid ordinary identifiers:
+42                    # Identifier does not start with alphabetic character
+with-numbers1234-5    # Embedded hyphen not followed by alphabetic character
+is-prime?             # Question mark is not alphanumeric
+x²                    # Superscript 2 has Unicode general category No
 =end code
 
-You use identifiers to name of constants, types (including classes and
-modules) and routines (subs and methods); they also appear in variable
-names usually proceeded by a sigil; see L<variables|/language/variables>
-for more details.
+In the C<x²> example above, the Unicode superscript numeral C<²> is not, and
+cannot be, a part of an ordinary identifier. Still, it may form part of a valid
+expression as an operator. The expression C<$x²>, for instance, produces the
+square of variable C<$x>. This should be somewhat surprising at this point,
+since operators are L<Subs|/type/Sub>, and, as mentioned above, Subs are named using
+identifiers. So how does the superscript numeral get accepted as (part of) an
+identifier? The key to this riddle lies in the concept of I<extended
+identifiers>, which is elaborated upon in the next section.
+
+=head3 Extended identifiers
+
+It is often convenient to have names that contain arbitrary characters.
+Use cases include situations where a set of entities shares a common "short"
+name, but still needs for each of its elements to be identifiable individually.
+For example, you might use a module whose short name is C<ThatModule>, but the
+complete long name of a module includes its version, naming authority, and
+perhaps even its source language. Similarly, sets of operators work together in
+various syntactic categories with names like C<prefix>, C<infix>, C<postfix>,
+etc. The (long) names of these operators, however, often contain characters that
+are excluded from ordinary identifiers.
```

Comment:
Instead  of long, which is imprecise, and maybe even wrong (² is not long), express what you mean. As a matter of fact, it might be the other way round: the identifier you declare those operators with include ordinary characters...

Reply:
OK, I see this is taken from the synopse. It's hard to say it's wrong then, but, well, maybe it can be expressed in some other way (also: if you take paragraphs from the synopse it would be better if you quote it explicitly)

Reply:
I tried to elucidate the text by including examples.